### ISSUE TITLE
Create function for sampling when aggregate deductibles/limits are present

### ISSUE THREAD

**Issue by ChrisWaller26** [ChrisWaller26/brms_freq_sev#43]
Currently, the model structure works when deductibles and limits are on a claim-by-claim basis. However, there may be aggregate features, such as annual aggregate deductibles and limits, which could also have a big impact.

This is much more difficult to model, as it requires multiple policies/claims to be grouped  together and so a key assumption in the model that all claims are independent of each other is violated.

It is possible to apply Monte Carlo simulations in `generated quantities` after the model has been fit to create samples which adjust for aggregate deductibles and limits but this will not impact the model fit itself.

A simple example of this is available here:

https://github.com/ChrisWaller26/Stan/blob/master/Pricing_Model/Pricing_Model.stan

**Comment by Atan1988** [comment]
Yes, this is definitely an interesting aspect of it.  Should we think about the trade-off between generated quantities vs something implemented via tidybayes package based on the posterior parameters space? I remembered that a few years ago in a stan meeting in NYC, I asked the stan dev team whether it would be possible to run only the generated quantities block without rerunning the model.  Some of these frequency severity model probably would take long time to run the bayesian sampling in real world applications.
